specs/interop/shared-lockbox.md

```diff
@@ -0,0 +1,123 @@
+# Shared Lockbox
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Design](#design)
+  - [Interface and properties](#interface-and-properties)
+  - [Events](#events)
+- [Reference implementation](#reference-implementation)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+With interoperable ETH, withdrawals will fail if the referenced `OptimismPortal` lacks sufficient ETH.
+The `SharedLockbox` improves the Superchain's interoperable ETH withdrawal user experience and avoids this issue.
+To do so, it unifies ETH L1 liquidity in a single contract (`SharedLockbox`), enabling seamless withdrawals of ETH
+from any OP chain in the Superchain, regardless of where the ETH was initially deposited.
+
+## Design
+
+The `SharedLockbox` contract is designed to manage the unified ETH liquidity for the Superchain.
+It implements two main functions: `lockETH` for depositing ETH into the lockbox,
+and `unlockETH` for withdrawing ETH from the lockbox.
+These functions are called by the `OptimismPortal` contracts to manage the shared ETH liquidity
+when making deposits or finalizing withdrawals.
+
+### Interface and properties
+
+**`lockETH`**
+
+Deposits and locks ETH into the lockbox's liquidity pool.
+
+- The function MUST accept ETH.
+- Only authorized `OptimismPortal` addresses SHOULD be allowed to interact.
+- The function MUST emit the `ETHLocked` event with the `portal` that called it and the `amount`.
+
+```solidity
+function lockETH() external payable;
+```
+
+**`unlockETH`**
+
+Withdraws a specified amount of ETH from the lockbox's liquidity pool.
+
+- Only authorized `OptimismPortal` addresses SHOULD be allowed to interact.
+- The function MUST emit the `ETHUnlocked` event with the `portal` that called it and the `amount`.
+
+```solidity
+function unlockETH(uint256 _value) external;
+```
+
+**`authorizePortal`**
+
+Grants authorization to a specific `OptimismPortal` contract.
+
+- Only `SuperchainConfig` address SHOULD be allowed to interact.
+- The function MUST add the specified address to the mapping of authorized portals.
+- The function MUST emit the [`AuthorizedPortal`](#events) event when a portal is successfully added.
+
+```solidity
+function authorizePortal(address _portal) external;
+```
+
+### Events
+
+**`ETHLocked`**
+
+MUST be triggered when `lockETH` is called
+
+```solidity
+event ETHLocked(address indexed portal, uint256 amount);
+```
+
+**`ETHUnlocked`**
+
+MUST be triggered when `unlockETH` is called
+
+```solidity
+event ETHUnlocked(address indexed portal, uint256 amount);
+```
+
+**`AuthorizedPortal`**
```

Comment:
To be consistent with the other events in this contract, i feel like it should be called `PortalAuthorized` since on the ETHLocked and ETHUnlocked we have `SUBJECT` first and `ADJECTIVE` later to describe the state of the subject. Here it would be the same "Portal" + "Authorized"